src/components/Users/UpdateProfileComponent.jsx

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react';
+import { Input, Row, Button } from 'react-materialize';
+import { Link } from 'react-router-dom';
+import PropTypes from 'proptypes';
+import uploader from '../../utils/uploader.util';
+import Navbar from '../Navbar';
+
+class UpdateProfileComponent extends Component {
+  state = {
+    username: '',
+    bio: '',
+    imageUrl: 'https://res.cloudinary.com/dbk8ky24f/image/upload/v1542099331/ezdrqhphpprculkmpc6g.png',
+  }
+
+  constructor(props) {
+    super(props);
+    this.fileInput = React.createRef();
+  }
+
+  componentDidUpdate(prevProps) {
+    const { success, history } = this.props;
+    if (!prevProps.success && success) {
+      if (success) {
+        history.push('/user-articles');
+      }
+    }
+  }
+
+  handleChange = (event) => {
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  fileSelectHandler = (event) => {
+    const selectedFile = event.target.files[0];
+    uploader({
+      image: selectedFile,
+      progress: () => {},
+      onSuccess: (url) => {
+        this.setState({
+          image_url: url,
+        });
+      },
+    });
+  }
+
+  onEdit = (event) => {
+    event.preventDefault();
+    const { updateProfileAction } = this.props;
+    updateProfileAction(this.state);
+  }
+
+  render() {
+    const { errors } = this.props;
+    const { username, bio, imageUrl } = this.state;
+    return (
+      <div>
+        <Navbar />
+        <div className="container">
+          <Row>
+            <form s={6} onSubmit={this.onEdit}>
+              <div className="user-data">
+                <div className="user-image">
+                  {/* eslint-disable-next-line */}
+                  <img src={imageUrl} alt={username} onClick={() => this.fileInput.click()} />
+                </div>
+                <input
+                  type="file"
+                  onChange={this.fileSelectHandler}
+                  className="file-input"
+                  ref={(fileInput) => { this.fileInput = fileInput; }}
+                />
+              </div>
+              <Row>
+                <Input
+                  name="username"
+                  placeholder="Ramon"
```

Comment:
@Ramon3162 Am not sure about this placeholder. What do you think?

Reply:
It shall be changed my brother, still working on this.